.github/workflows/reusable-release.yml

```diff
@@ -342,25 +343,32 @@ jobs:
         run: |
           echo "KEY_FINGERPRINT=$KEY_FINGERPRINT"
           find $LOCAL_ARTIFACTS_STAGING_PATH -type f -exec gpg -a --local-user "$KEY_FINGERPRINT" --detach-sign {} \;
+      - name: "Full Main-Branch release: Add release commit (from temporary release branch) to default branch"
+        if: needs.init.outputs.release_type == 'FULL_MAIN_BRANCH'
+        env:
+          GH_TOKEN: ${{ steps.generate-github-app-token.outputs.token }}
+          GH_REPO: ${{ github.repository }}
+          RELEASE_COMMIT_ID: ${{ needs.push-release-commit.outputs.release_commit_id }}
+        run: |
+          if gh api --silent --method PATCH /repos/:owner/:repo/git/refs/heads/$GITHUB_REF_NAME -f "sha=$RELEASE_COMMIT_ID"; then
+            echo "...fast-forward of default branch to include release commit succeeded"
```

Comment:
[`PATCH /repos/:owner/:repo/git/refs/heads/{branch_name}`](https://docs.github.com/en/rest/git/refs?apiVersion=2022-11-28#update-a-reference) is the GitHub API for updating a branch to point at a new commit - it will only allow fast-forwards, and not rewriting history, so long as you don't use the `force` param - and we don't.

Previously we